src/components/content/observations/patient-observations-outside.tsx

```diff
@@ -1,18 +1,26 @@
 import cx from "classnames";
+import { useEffect } from "react";
 import { PatientObservationsBase } from "@/components/content/observations/helpers/base";
 import { withErrorBoundary } from "@/components/core/error-boundary";
 import { usePatientDiagnosticReportsOutside } from "@/fhir/diagnostic-report";
+import { Telemetry } from "@/utils/telemetry";
 
 export type PatientObservationsOutsideProps = {
   className?: cx.Argument;
 };
 
 const Component = ({ className }: PatientObservationsOutsideProps) => {
-  const diagnosticReports = usePatientDiagnosticReportsOutside();
+  const query = usePatientDiagnosticReportsOutside();
+
+  useEffect(() => {
+    if (!query.isLoading) {
+      Telemetry.reportZAPRecordCount("outside_observations", query.data?.length);
+    }
+  }, [query.isLoading, query.data]);
```

Comment:
same here (checking query.data exists before calling reportZAPRecordCount

```suggestion
  useEffect(() => {
    if (!query.isLoading && query.data) {
      Telemetry.reportZAPRecordCount("outside_observations", query.data.length);
    }
  }, [query.isLoading, query.data]);
```